app/code/Magento/Catalog/Controller/Adminhtml/Product/MassDelete.php

```diff
@@ -32,38 +41,55 @@ class MassDelete extends \Magento\Catalog\Controller\Adminhtml\Product implement
      */
     private $productRepository;
 
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
     /**
      * @param Context $context
      * @param Builder $productBuilder
      * @param Filter $filter
      * @param CollectionFactory $collectionFactory
      * @param ProductRepositoryInterface $productRepository
+     * @param LoggerInterface $logger
      */
     public function __construct(
         Context $context,
         Builder $productBuilder,
         Filter $filter,
         CollectionFactory $collectionFactory,
-        ProductRepositoryInterface $productRepository = null
+        ProductRepositoryInterface $productRepository = null,
+        LoggerInterface $logger = null
     ) {
         $this->filter = $filter;
         $this->collectionFactory = $collectionFactory;
-        $this->productRepository = $productRepository
-            ?: \Magento\Framework\App\ObjectManager::getInstance()->create(ProductRepositoryInterface::class);
+        $this->productRepository = $productRepository ?:
+            \Magento\Framework\App\ObjectManager::getInstance()->create(ProductRepositoryInterface::class);
+        $this->logger = $logger ??
```

Comment:
@edenduong Minor change: The line above this uses ternary operator (?:) for the same purpose, can you use same here? Let me know if I am missing something. Thanks for the contribution!

Reply:
Thank you @kalpmehta . I have changed the source code. Thanks!